contributing/principles.md

```diff
@@ -0,0 +1,90 @@
+language. In practice, this means:
+
+- Publicly exposed components in FDS should be named based on how design refers to the
+component in their sketch library
+- Component structure and semantic grouping should reflect design intention
+
+When someone refers to a "Chip", we should have a shared understanding across design,
+engineering, and product.
+
+## Small, sharp tools
+Components, CSS classes, and other tools in FDS follow what could be called the ["Unix
+Philosphy"](https://en.wikipedia.org/wiki/Unix_philosophy). There is no formal definition, but this summary fits well:
+
+> Write programs that do one thing and do it well. Write programs to work together.
+
+Consider the design of `cat`:
+
+> The design of cat is typical of most UNIX programs: it implements one simple but general function that can be used in many different applications (including many not envisioned by the original author).
+
+Making our CSS classes and React components as small and sharp as possible gives consumers
+the flexibility to compose those smaller pieces into more complex patterns.
+
+Each individual component or class should behave the same way no matter where it lives in a
+consumer application.
+
+Each component should define only the props it needs for presentation.
+
+Each component or utility class should strive to do one, and only one thing.
+
+
+## "Worse is Better"
+Building a generic component that can accommodate all future use cases is hard. To
+optimize for building components and gaining contributors, we can ease the burden of
+designing a generic component by following the ["Worse is Better"](https://en.wikipedia.org/wiki/Worse_is_better) method.
+
+Here are some key points of this method as it applies to FDS:
+- Simplicity is the most important consideration in a design.
+- Components should be consistent, **but not at the sacrifice of simplicity**.
+- Components should cover as many use cases as possible, but **completeness is not as important as simplicity**
+
+This means that a component only needs to support the absolute minimum functionality before shipping.
+
+**Favor software designs that can be iterated on as needed instead of optimizing for complexity up-front.**
+
+## Native web standards preferred
+You can do almost anything in javascript, but that doesn't mean you should. Components
+should leverage as much as possible from native web standards supported by modern
+browsers.
+
+It is unlikely that a custom JS solution will perform as reliably as something implemented
+natively in a browser.
+
+For example, `position: sticky` should be preferred over a JS solution to fixing content
+during scroll.
+
+### JS-in-JS, CSS-in-CSS
+In FDS, we have opted for keeping styling as close to native web standards as possible.
+We do not use any CSS-in-JS libraries. We do not use CSS preprocessors. CSS is written
+based on the latest CSS standards (with help from PostCSS for polyfills and vendor sugar).
```

Comment:
I think PostCSS in a sense IS a preprocessor (polyfills, vendor). You could technically enable Sass like features pretty easily with it.

Reply:
I'm currently working on a "CSS in JS" article that _could_ be a candidate for a good blog post. If so, we could drop a link to that right in here.

Reply:
Editing this to clear up what I actually mean by "preprocessors"